src/ai/backend/manager/models/scaling_group.py

```diff
@@ -166,24 +166,68 @@ def as_trafaret(cls) -> t.Trafaret:
 )
 
 
+@overload
 async def query_allowed_sgroups(
     db_conn: SAConnection,
     domain_name: str,
-    group: Union[uuid.UUID, str],
+    group: uuid.UUID,
+    access_key: str,
+) -> Sequence[Row]:
+    ...
+
+
+@overload
+async def query_allowed_sgroups(
+    db_conn: SAConnection,
+    domain_name: str,
+    group: Iterable[uuid.UUID],
+    access_key: str,
+) -> Sequence[Row]:
+    ...
+
+
+@overload
+async def query_allowed_sgroups(
+    db_conn: SAConnection,
+    domain_name: str,
+    group: str,
+    access_key: str,
+) -> Sequence[Row]:
+    ...
+
+
+@overload
+async def query_allowed_sgroups(
+    db_conn: SAConnection,
+    domain_name: str,
+    group: Iterable[str],
+    access_key: str,
+) -> Sequence[Row]:
+    ...
+
+
+async def query_allowed_sgroups(
+    db_conn: SAConnection,
+    domain_name: str,
+    group: uuid.UUID | Iterable[uuid.UUID] | str | Iterable[str],
     access_key: str,
 ) -> Sequence[Row]:
     query = sa.select([sgroups_for_domains]).where(sgroups_for_domains.c.domain == domain_name)
     result = await db_conn.execute(query)
     from_domain = {row["scaling_group"] for row in result}
 
-    group_id = await resolve_group_name_or_id(db_conn, domain_name, group)
+    if isinstance(group, Iterable):
+        group_ids = await resolve_groups(db_conn, domain_name, group)
+    else:
+        group_id = await resolve_group_name_or_id(db_conn, domain_name, group)
+        if group_id is None:
+            group_ids = []
+    group_cond = sgroups_for_groups.c.group.in_(group_ids)
     from_group: Set[str]
-    if group_id is None:
+    if not group_ids:
         from_group = set()  # empty
     else:
-        query = sa.select([sgroups_for_groups]).where(
-            (sgroups_for_groups.c.group == group_id),
-        )
+        query = sa.select([sgroups_for_groups]).where(group_cond)
```

Comment:
let `query_allowed_sgroups()` accept list-shaped group parameter.
Then it is able to query allowed scaling groups by access key.
https://github.com/lablup/backend.ai/blob/b083e33c5a0577eecc691194f8e1fdd26569d54a/src/ai/backend/manager/models/agent.py#L374